packages/stack/src/views/Stack/CardContainer.tsx

```diff
@@ -120,11 +129,22 @@ function CardContainer({
     onCloseRoute({ route: scene.route });
   };
 
+  const handleGestureBegin = () => {
+    onPageChangeStart?.();
+    onGestureStart?.({ route: scene.route });
+  };
+
+  const handleGestureCanceled = () => {
+    onPageChangeCancel?.();
+    onGestureCancel?.({ route: scene.route });
+  };
+
   const handleTransitionStart = ({ closing }: { closing: boolean }) => {
     if (active && closing) {
       onPageChangeConfirm?.();
+      onGestureEnd?.({ route: scene.route });
```

Comment:
`onGestureEnd` needs to explicitly be called on gesture end. transition could start due to various reasons, and doesn't mean that a gesture has ended.

Reply:
Ok, removing `onGestureEnd` call here, call back will be invoked from `Card.tsx`